lib/hci-socket/smp.js

```diff
@@ -28,14 +29,21 @@ var Smp = function(aclStream, localAddressType, localAddress, remoteAddressType,
 
   this._aclStream.on('data', this.onAclStreamDataBinded);
   this._aclStream.on('end', this.onAclStreamEndBinded);
+
+  this._remoteAddress = remoteAddress;
```

Comment:
Is this._remoteAddress necessary, when we already have this._ra, which is at least somewhat similar?

Reply:
this._ra is the raw binary data that gets passed into the crypto function, and is not really useful for passing along to passkeyProvider().  We could convert it back from the buffer when we need it, but that seems like more trouble than it's worth when we have (almost) what we need already in the remoteAddress parameter.